bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -112,19 +151,121 @@ boolean isRunning() {
     void suspendProcessing() {
         synchronized (suspensionLock) {
             suspended = true;
-            allChannels.setReadable(false).awaitUninterruptibly();
+            for (Channel channel : allChannels) {
+                channel.config().setAutoRead(false);
+            }
         }
     }
 
     @VisibleForTesting
     void resumeProcessing() {
         synchronized (suspensionLock) {
             suspended = false;
-            allChannels.setReadable(true).awaitUninterruptibly();
+            for (Channel channel : allChannels) {
+                channel.config().setAutoRead(true);
+            }
             suspensionLock.notifyAll();
         }
     }
 
+    private void listenOn(InetSocketAddress address, BookieSocketAddress bookieAddress) throws InterruptedException {
+        if (!conf.isDisableServerSocketBind()) {
```

Comment:
@revans2 in  BOOKKEEPER-930 we refactored the code in order to have a more clear concept of ChannelManager. I think it would be better to leave the code in the form of BOOKKEEPER-930 which is more modular as @sijie proposed at the time of implementing  BOOKKEEPER-930